src/postamble.js

```diff
@@ -398,11 +398,7 @@ function exit(status, implicit) {
   EXITSTATUS = status;
 
 #if ASSERTIONS && !EXIT_RUNTIME
-  // Skip this check if the runtime is being kept alive deliberately.
-  // For example if `exit_with_live_runtime` is called.
-  if (!runtimeKeepaliveCounter) {
```

Comment:
why remove the condition here?

Reply:
The `runtimeKeepaliveCounter` is not available here.. since `EXIT_RUNTIME=0` in this block of code.   I think its not very interesting since users of `exit_with_live_runtime` probably have `EXIT_RUNTIME=1` set.

I suppose the downside of removing this check is that more programs with unflushed content on `exit()` could see this warning.   I'm not sure how serious that is?

Reply:
I think a problem is that the application might have buffered data, and end up showing this error, but the program is still running so it gets flushed later, which would be confusing. Basically we only want that warning to be shown when the program is definitely shutting down. Given that, perhaps it could be moved to later in this function?

(Regardless I think we should keep show the warning both with and without EXIT_RUNTIME, since `EXIT_RUNTIME=0` is the default and most users benefit from it there.)

Reply:
It might not be worth it, I see what you mean. But it's hard to say how many false positives we'll get here...

If we get bug reports, would you be willing to revert this or re-add `runtimeKeepaliveCounter` in `ASSERTIONS` mode?

Reply:
I'm fairly sure this won't effect many/any folks.

In order to be effect you would need to:

(a) Exit with the runtime still alive (i.e. set a main loop) and (b) Have partial output in stdout (i.e. printf without a newline at the end).

What is more, for a very long time the call to checkUnflushedContent was completely unguarded.   The guard on it was not introduced until very recently in #16140 (Jan 2022)

Reply:
I agree it's not likely. But if/when it happens, it might be confusing. I guess it's just in assertions builds, though, so sgtm...